plugins/chat/assets/javascripts/discourse/components/chat/navbar/back-button.gjs

```diff
@@ -12,6 +12,18 @@ export default class ChatNavbarBackButton extends Component {
     return this.args.title ?? I18n.t("chat.browse.back");
   }
 
+  get targetRoute() {
+    if (
+      ["chat.direct-messages", "chat.channels", "chat.threads"].includes(
+        this.args.route
+      )
+    ) {
+      return this.args.route;
+    } else {
+      return "chat";
+    }
```

Comment:
I wonder if `[...].some(route => route.startsWith(this.args.route))` wouldn't be more resilient?

Reply:
It's just that in some cases you can have: "chat.channels.index" for example